src/main/java/com/aws/greengrass/shadowmanager/sync/IotDataPlaneClientWrapper.java

```diff
@@ -57,10 +58,12 @@ public void setRate(int rate) {
      * @param shadowName The shadow name associated with the sync shadow update
      * @throws IoTDataPlaneClientCreationException when the iot data plane client is not created
      */
+    @SuppressWarnings("PMD.CloseResource")
     public DeleteThingShadowResponse deleteThingShadow(String thingName, String shadowName)
             throws IoTDataPlaneClientCreationException {
+        IotDataPlaneClient client = iotDataPlaneClientFactory.getIotDataPlaneClient();
         rateLimiter.acquire();
-        return iotDataPlaneClientFactory.getIotDataPlaneClient().deleteThingShadow(DeleteThingShadowRequest.builder()
+        return client.deleteThingShadow(DeleteThingShadowRequest.builder()
```

Comment:
Is this even worth doing? If we can't get a data client, then the rate limiter doesn't matter

Reply:
Yes, with this, exception is thrown without acquiring rate limiter.

Reply:
It's just more code without any benefit though? I think this can be backed out